saleor/graphql/order/mutations/orders.py

```diff
@@ -590,6 +595,12 @@ def perform_mutation(cls, _root, info, **data):
         invoice.order = instance
         invoice.status = InvoiceStatus.READY
         invoice.save()
+        events.invoice_created_event(
+            user=info.context.user,
+            invoice=invoice,
+            number=data["input"].get("number"),
+            url=data["input"].get("url"),
+        )
         return CreateInvoice(invoice=invoice)
```

Comment:
The current solution allows executing mutations on draft orders. Should we allow this?

Reply:
It's a good point, I thing generating an invoice should only be allowed for fullfilled order.

Reply:
@maarcingebala For witch orders, we want to create invoices?

Reply:
For all orders except drafts? If an order was placed successfully, the customer should receive the invoice. If we can make any changes to orders we can also generate new invoices.

Reply:
We should add the same validation to `RequestInvoice`